src/ncp/spinel.h

```diff
@@ -33,6 +33,17 @@
 #ifndef SPINEL_HEADER_INCLUDED
 #define SPINEL_HEADER_INCLUDED 1
 
+/*
+ * Spinel definition guideline:
+ *
+ * New NCP firmware should work with an older host driver, i.e., NCP implementation should remain backward compatible.
+ *
+ *  - Existing fields in the format of an already implemented spinel property or command cannot change.
+ *  - New fields may be appended at the end of the format (or the end of a struct) as long as the NCP implementation
+ *    treats the new fields as optional (i.e., a driver not aware of and therefore not using the new fields should
+ *    continue to function as before).
```

Comment:
Also allow renaming properties?

Reply:
We do allow it but it's not encouraged. So I am not sure if we want to explicitly mention it here. 
Thoughts?

Reply:
Got it!